chsdi/views/faqlist.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+import requests
+from requests.exceptions import RequestException
+from pyramid.view import view_config
+from pyramid.httpexceptions import (HTTPNotFound, HTTPBadRequest)
+
+
+@view_config(route_name='faqlist', renderer='json')
+def faqlist(self):
+    try:
+        layersConfigURL = self.route_url('layersConfig', map='api')
+    except Exception:
+        raise HTTPNotFound()
+
+    try:
+        layers_api = requests.get(layersConfigURL).json()
+    except Exception:
+        raise HTTPBadRequest()
+
+    suffix3d = '_3d'
+    queryable_layers = []
+    searchable_layers = []
+    chargeable_layers = []
+    no_chargeable_layers = []
+
+    for layer_name in layers_api:
+        try:
+            layer = layers_api[layer_name]
+            if 'parentLayerId' not in layer and not str(layer_name).endswith(suffix3d):
+                if 'queryable' in layer and layer['queryable']:
+                    try:
+                        queryable_layers.append(layer_name)
+                    except:
+                        pass
+                if 'searchable' in layer and layer['searchable']:
+                    try:
+                        searchable_layers.append(layer_name)
+                    except:
+                        pass
+                if 'chargeable' in layer and layer['chargeable']:
+                    try:
+                        chargeable_layers.append(layer_name)
+                    except:
+                        pass
+                if 'chargeable' in layer and not layer['chargeable']:
+                    try:
+                        no_chargeable_layers.append(layer_name)
+                    except:
+                        pass
+        except RequestException:
+            pass
+
+    if (len(queryable_layers) + len(searchable_layers) + len(chargeable_layers) + len(no_chargeable_layers)) == 0:
+        raise HTTPBadRequest()
```

Comment:
why is that? `HTTPBadRequest` should be not thrown here. You can use that when parameters are bad/wrong.